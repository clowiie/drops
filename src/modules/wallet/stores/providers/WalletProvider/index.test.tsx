import { render } from '@testing-library/react'

import * as WalletProviderType from '.'

describe('WalletProvider', () => {
  const useInitWalletSpy = jest.fn()
  jest.doMock('modules/wallet/hooks/useInitWallet', () => useInitWalletSpy)

  const Component = () => <h1>Component</h1>

  const { default: WalletProvider } = require('.') as typeof WalletProviderType

  beforeEach(() => {
    useInitWalletSpy.mockReturnValue({
      walletAddress: 'address',
    })
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('snapshots', () => {
    it('should match snapshots', () => {
      const { container, getByText } = render(
        <WalletProvider>
          <Component />
        </WalletProvider>,
      )

      expect(container.firstChild).toMatchSnapshot()
      expect(getByText('Component')).toBeDefined()
    })
  })
})
