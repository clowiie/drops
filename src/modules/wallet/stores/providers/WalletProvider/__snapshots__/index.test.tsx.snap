// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletProvider snapshots should match snapshots 1`] = `
<h1>
  Component
</h1>
`;
